### ISSUE TITLE
Can `datalad push` also be made to push tags?

### ISSUE THREAD

**Issue by mih** [psychoinformatics-de/knowledge-base#49]
Origin: DataLad chatroom 2023-05-02

Can `datalad push` also be made to push tags?

TODO (not necessarily to be performed in this order)
- [ ] Inform OP/Add reference to this issue at origin
- [x] Clarifying Qs asked or not needed
- [x] Nature of the issue is understood
- [x] Inform OP about resolution

**Comment by mih** [comment]
`datalad push` is meant to work similar to `git push` and, importantly, honor its configuration. So the following is a demo how pushing tags works:

```
# set up a demo dataset
❯ datalad create ds
# demo git remote
❯ mkdir remote
❯ git -C remote init --bare
# register the remote in the dataset
❯ datalad siblings -d ds add -s origin --url /tmp/push/remote/
# configure all branches to be pushed by default
❯ git -C ds config --add remote.origin.push 'refs/heads/*'
# configure all tags to be pushed by default too
❯ git -C ds config --add remote.origin.push 'refs/tags/*'
# add a tag to the dataset
❯ git -C ds tag mytag
# push
❯ datalad push -d ds --to origin
publish(ok): . (dataset) [refs/heads/git-annex->origin:refs/heads/git-annex [new branch]]                      
publish(ok): . (dataset) [refs/heads/master->origin:refs/heads/master [new branch]]                            
publish(ok): . (dataset) [refs/tags/mytag->origin:refs/tags/mytag [new tag]]                                   
```

**Comment by mih** [comment]
Fix in #52 but was not closed
